doc/API/README.md

```diff
@@ -0,0 +1,3 @@
+# CProver APIs
```

Comment:
This should be changed to something like:

```
\page apis CPROVER APIs
```

Reply:
Done.  I've taken the liberty of converting `doc/API` and `doc/ADR` to the doxygen page/subpage syntax.  I was going to do the same for `doc/assets`, but there is an xml-spec.md that is not linked to by the top-level README, and I wasn't sure which of the various specs are considered authoritative.  @NlightNFotis